docfx/doxygen2syntax.cc

```diff
@@ -101,6 +117,94 @@ void TemplateParamListSyntaxContent(std::ostream& os,
   os << ">\n";
 }
 
+std::string ReturnDescription(YamlContext const& /*ctx*/,
+                              pugi::xml_node const& node) {
+  // The return description, if present, is in a `<simplesect>` node that is
+  // part of the *function* description.
+  auto selected = node.select_node(".//simplesect[@kind='return']");
+  if (!selected) return {};
+  std::ostringstream os;
+  MarkdownContext mdctx;
+  mdctx.paragraph_start = "";
+  AppendDescriptionType(os, mdctx, selected.node());
+  return std::move(os).str();
+}
+
+// We need to search the parameters in the `<parameterlist>` element. The type
+// of this element is defined as below. Note that this is basically a sequence
+// of "parameter items". The "parameter items" contain the description, and
+// may contain a *list* of parameter names.
+//
+// clang-format off
+//   <xsd:complexType name="docParamListType">
+//     <xsd:sequence>
+//       <xsd:element name="parameteritem" type="docParamListItem" minOccurs="0" maxOccurs="unbounded" />
+//     </xsd:sequence>
+//     <xsd:attribute name="kind" type="DoxParamListKind" />
+//   </xsd:complexType>
+//   <xsd:complexType name="docParamListItem">
+//     <xsd:sequence>
+//       <xsd:element name="parameternamelist" type="docParamNameList" minOccurs="0" maxOccurs="unbounded" />
+//       <xsd:element name="parameterdescription" type="descriptionType" />
+//     </xsd:sequence>
+//   </xsd:complexType>
+//   <xsd:complexType name="docParamNameList">
+//     <xsd:sequence>
+//       <xsd:element name="parametertype" type="docParamType" minOccurs="0" maxOccurs="unbounded" />
+//       <xsd:element name="parametername" type="docParamName" minOccurs="0" maxOccurs="unbounded" />
+//     </xsd:sequence>
+//   </xsd:complexType>
+// clang-format on
+bool ParameterItemMatchesName(std::string_view parameter_name,
+                              pugi::xml_node const& item) {
+  for (auto const& list : item.children("parameternamelist")) {
+    for (auto const& name : list.children("parametername")) {
+      if (std::string_view{name.child_value()} == parameter_name) return true;
+    }
+  }
+  return false;
+}
+
+std::string ParameterDescription(YamlContext const& /*ctx*/,
+                                 pugi::xml_node const& node,
+                                 std::string_view parameter_name) {
+  // The parameter descriptions, if present, is in a `<simplesect>` node that is
```

Comment:
s/descriptions/description/ OR s/descriptions is/descriptions are/?

Reply:
Fixed